### PR TITLE
chore(kv): refactoring store with reusable store type

### PR DESCRIPTION
**Pull request by jsteenb2**
this relates to #16316 

the main gist of this work is to provide some more rigidity around store access. Its a bit all over the place atm. This aims to make the 95% use case completely repeatable, and allow other stores to compose this into even more full fledged solutions

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] update upstream tests to pass with slightly changed error messages

### CODE REVIEW

File: kv/notification_endpoint.go [hunk 1]
```diff
@@ -116,7 +74,26 @@ func (s *Service) createNotificationEndpoint(ctx context.Context, tx Tx, edp inf
 	edp.SetUpdatedAt(now)
 	edp.BackfillSecretKeys()
 
-	if err := s.putNotificationEndpoint(ctx, tx, edp); err != nil {
+	if err := edp.Valid(); err != nil {
```
Comment (jsteenb2) [inline kv/notification_endpoint.go]: this validation will eventually get moved into service layer in #16316

File: kv/document_test.go [hunk 2]
```diff
@@ -6,16 +6,6 @@ import (
 	influxdbtesting "github.com/influxdata/influxdb/testing"
 )
 
-func TestBoltDocumentStore(t *testing.T) {
```
Comment (kelwang) [inline kv/document_test.go]: looks like the inmem test for this one is skipped

Comment (jsteenb2) [reply kv/document_test.go]: same with the notificaiton rules store oddly enough. No idea what all those are about tbh 🤷‍♂.

Comment (kelwang) [reply kv/document_test.go]: not a big deal, we will build the inmem transaction in another pr

**Comment by jsteenb2** [comment]
upstream tests failing, already have fix upstream ready to merge

**Pull request merged**